File: backend/PluriConnect_Api/Database/ActivityInstance.cs
using System;
using System.Collections.Generic;

namespace PluriConnect_Api.Database;

public partial class ActivityInstance
{
    public int Id { get; set; }

    public int ChildId { get; set; }

    public int ActivityId { get; set; }

    public string AssignedAt { get; set; } = null!;

    public int IsAuthorized { get; set; }

    public string? AuthorizedAt { get; set; }

    public virtual Activity Activity { get; set; } = null!;

    public virtual Child Child { get; set; } = null!;

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();
}

File: Client/src/types/ActivityInstance.ts
import type { ActivityDetailsType } from './ActivityDetailsType';
import type { childDetails } from './childDetails';
export interface ActivityInstanceType {
  Id: number;
  ChildId: number;
  ActivityId: number;
  AssignedAt: string;
  IsAuthorized: number;
  AuthorizedAt?: string | null;

  Activity: ActivityDetailsType;
  Child: childDetails;
  Comments: Comment[];
}